[Docs] Guide for self-hosting vs cloud-hosting (#7114)
medplum/medplum: packages/docs/docs/self-hosting/considerations.md

Reviewer: I think content is generally good here, we want to tweak the tone to be a bit more matter-of-fact, rather than dissuasive.

---
sidebar_position: 1
---

> # Self-hosting vs Cloud

Author: ```suggestion
# Self-Hosting vs Cloud
```

---
sidebar_position: 1
---

# Self-hosting vs Cloud

Self-hosting Medplum can be the right choice for certain use cases, but it comes with significant operational overhead that many organizations underestimate. This guide helps you evaluate whether self-hosting aligns with your needs and resources.

## When Self-hosting Makes Sense

Self-hosting may be appropriate if you have one of these compelling reasons:

**Compliance Requirements**
You need data sovereignty or have contractual obligations that require on-premises deployment, such as EU regulations or US federal government work. 

**On-Device or Small On-Prem FHIR Server**

> You might need to run a FHIR server on a medical device you are developing, or on a small machine for a rural clinic, and a lightweight Medplum instance with minimal interactions is sufficient. 

Reviewer: ```suggestion
Your project has limited data storage / bandwidth requirements and operates in a low-connectivity environment. You might need to run a FHIR server on a medical device you are developing, or on a small machine for a rural clinic, and a lightweight Medplum instance with minimal interactions is sufficient. 
```

> **Learning Project**

Reviewer: ```suggestion
**Personal Education**
```

You want to gain experience with self-hosting infrastructure, and your Medplum instance will only be used for personal projects without scale requirements.

**Fork Requirements**

> You have a [highly compelling technical reason](/blog/so-youre-thinking-about-forking) to fork the Medplum codebase that cannot be addressed through configuration or our extension points.

Author: removed

## Hidden Costs of Self-hosting

Before choosing self-hosting, carefully consider these often-overlooked operational costs:

> **Ongoing Maintenance Burden**

Reviewer: ```suggestion
**Ongoing Maintenance**
```

From our experience, you'll need at least 0.5 FTE (full-time equivalent) dedicated to maintaining your Medplum instance during normal quarters. This includes general maintenance for system stability and ensuring you stay current with version upgrades—an area where many self-hosting customers struggle due to competing priorities and loss of organizational knowledge. 

> During quarters requiring *large-scale maintenance projects* like PostgreSQL version upgrades or major Medplum version upgrades, resource requirements can scale to 3+ FTEs.

Reviewer 2: This can vary a ton depending on the organization and the workload.  3+ FTE would be the high end.  We don't want to be alarmist with these estimates.

**Operational Responsibilities**

> Your team will take on 24/7 pager duty for outages and need to build incident response protocols, implement observability in your cluster, and maintain all the supporting infrastructure that comes with production systems.

Reviewer: ```suggestion
Your team will take on 24/7 on-call responsibility for outages and need to build incident response protocols, implement observability systems, and maintain all the supporting infrastructure that comes with production systems.
```

## What Medplum Cloud-Hhosting Provides

**Managed Operations**

> We handle all system maintenance, security updates, and infrastructure scaling so your team can focus on building your healthcare application.

Reviewer: ```suggestion
Medplum handles all system maintenance, security updates, and infrastructure scaling so your team can focus on building your healthcare application.
```

**Log Streaming**
On our cloud-hosted Premium and Enterprise plans, seamless log streaming integration with common observability tools like SumoLogic and DataDog, giving you visibility into your application without managing the underlying infrastructure.

## Already Self-hosting at Scale?

> If you're reading this after already hitting scale and struggling under the operational weight of maintaining your servers, we're here to help. 

Reviewer: I'd think about removing the first part of the "if" that assumes problems.

"If you're a current self-hosting user that would like to transition to cloud hosting, we're here to help!"
